coins/coins.go

```diff
@@ -182,105 +201,172 @@ func (a *WeiAmount) BigInt() *big.Int {
 		panic(err)
 	}
 	if cond.Inexact() {
-		// We round when converting from Ether to Wei, so we shouldn't see this
+		// We round when converting from ETH to Wei, so we shouldn't see this
 		log.Warnf("converting WeiAmount=%s to big.Int required rounding", a.String())
 	}
 	return new(big.Int).SetBytes(wholeWeiVal.Coeff.Bytes())
 }
 
-// AsEther returns the wei amount as ether
+// AsEther returns the Wei amount as ETH
 func (a *WeiAmount) AsEther() *apd.Decimal {
 	ether := new(apd.Decimal).Set(a.Decimal())
 	decreaseExponent(ether, NumEtherDecimals)
 	_, _ = ether.Reduce(ether)
 	return ether
 }
 
-// AsEtherString converts the wei amount to an eth amount string
+// AsEtherString converts the Wei amount to an ETH amount string
 func (a *WeiAmount) AsEtherString() string {
 	return a.AsEther().Text('f')
 }
 
-// AsStandard is an alias for AsEther, returning the wei amount as ether
+// AsStandard is an alias for AsEther, returning the Wei amount as ETH
 func (a *WeiAmount) AsStandard() *apd.Decimal {
 	return a.AsEther()
 }
 
-// String returns the wei amount as a base10 string
+// AsStandardString is an alias for AsEtherString, returning the Wei amount as
+// an ETH string
+func (a *WeiAmount) AsStandardString() *apd.Decimal {
+	return a.AsEther()
+}
+
+// StandardSymbol returns the string "ETH"
+func (a *WeiAmount) StandardSymbol() string {
+	return "ETH"
+}
+
+// IsToken returns false, as WeiAmount is not an ERC20 token
+func (a *WeiAmount) IsToken() bool {
+	return false
+}
+
+// TokenAddress returns the all-zero address as WeiAmount is not an ERC20 token
+func (a *WeiAmount) TokenAddress() ethcommon.Address {
+	return ethcommon.Address{}
+}
+
+// String returns the Wei amount as a base10 string
 func (a *WeiAmount) String() string {
 	return a.Decimal().Text('f')
 }
 
-// FmtWeiAsETH takes wei as input and produces a formatted string of the amount
+// FmtWeiAsETH takes Wei as input and produces a formatted string of the amount
 // in ETH.
 func FmtWeiAsETH(wei *big.Int) string {
 	return NewWeiAmount(wei).AsEther().Text('f')
 }
 
+// ERC20TokenInfo stores the token contract address and basic info that most
+// ERC20 tokens support
+type ERC20TokenInfo struct {
+	Address     ethcommon.Address `json:"address" validate:"required"`
+	NumDecimals uint8             `json:"decimals"` // digits after the Decimal point needed for smallest denomination
```

Comment:
not required?

Reply:
Required would mean that the value can never be zero.

Reply:
I found an example: https://etherscan.io/address/0xaec2e87e0a235266d9c5adc9deb4b2e29b54d009#code
I don't think there are any ones that people would want to use though, if we decide to ban 0-decimal tokens.